### DISCUSSION TITLE
Permissions, Testing, and Best Practice

### DISCUSSION THREAD

**Discussion by kevinwcrump** [jongpie/NebulaLogger#611] [Q&A] [answered]
I have a few questions and concerns around the Permissions Sets and best approach in our new Nebula Logger implementation ( not yet  promoted to Production.
1) Is it true that all Users ( Community/Portal) must belong to one of the PermissionSets ( or have same permissions) for Apex to create a log and log entry(ies) when running in the context of that user?
2) If a call to the logging methods is encountered during execution and the User in context DOES NOT have one of the permissions or ability to create logs- does Nebula Logger just ignore the call? Will this also raise an exception?

3) I was able to Log from Apex running in the context of a Community User who IS NOT assigned 1 of the permission sets. Perhaps something configured wrong, or my assumptions are incorrect?

I am attempting to Test this myself in Unit Tests with runAs .. creating both Admin (who owns account) and Portal User but repeatedly hit the  'AccountOwner must have Role' error ( It does.) .. So having difficulty testing out my own assumptions. 

We have MANY portal  (Community Plus) users..   do we need to use Data Loader and assign them ALL to one of the permission sets?

I would appreciate any feedback and guidance or assurances you may have.

Other than this implementation /testing issue  we absolutely love Nebula Logger so far!

Thank you.

**Comment by jongpie** [comment, marked as answer]
Hi @kevinwcrump - I saw [your comment on #449](https://github.com/jongpie/NebulaLogger/discussions/449#discussioncomment-8212615), glad to hear that it had the info you needed! I'm adding the link here for future reference.

Just let me know if you run into any issues with using Nebula Logger within your Community site (or have any other questions).
